### ISSUE TITLE
[Developer Version] Bad [Developer Version]

### ISSUE THREAD

**Issue by AbominationBanisher** [Regalis11/Barotrauma#515]
After going through some hoops to get the version running, I instantly ran into features/bugs, that made me decide to avoid purchasing the game if they were kept.
-------------------------------------------------------------------------------------------------------------------
Problematic features:

The player HUD: It is too intrusive, unorganized, limits the player's free will, too clunky, confusing, slow, and is overall a pain in the ass. I'd like the original HUD, with only minor changes that you seem fit. But the current HUD, is in my opinion, disgusting.

Medical stuff: I don't like anything about the changes to the medic. Applying stuff to certain body parts, stuff like that, is just plainly not needed. I don't like, and to be honest, it is much too clunky.

Random HUD's: The HUD of the engine, of the reactor, and of everything else, is clunky, too extra, confusing, and most importantly, not needed. I find it to be terrible.

Metal crate: Just no, keep it the old way.

New machines: Are they necessary? No. If you're gonna add new things, do it in the style that barotrauma extended did them. ---------- [EDIT]: You just remodeled the status monitors. Anyways, the sprites are not good. ----------------- [DOUBLE EDIT]: Most of thing that were remodeled look terrible. Change them back to how they were before, please. --------- [TRIPLE EDIT]: EVERYTHING THAT YOU REMODELED LOOKS TERRIBLE. THE STATIONS, THE  BACKGROUND, THE TEXTURES OF THE SHIP, THE INSIDES OF THE STORAGES, THE MENU BUTTONS, JESUS. --------- [QUADRUPLE EDIT] THE TEXTURES ARE ONLY JACKSHIT ON POLARIS

The Polaris: It's like every single custom map, ever made: SO CLUNKY, IN YOUR FACE, AND GENERALLY TERRIBLE.

The sliding movement: The sliding movement for the batteries, water pumps, engine, and more, is not welcome. Its clunky.

Curved insides: I don't like how inside the sub, you can go up on the sloped parts. It does not feel right, and makes the submarine generally feel unorganized and clunky, which as I mentioned before, I don't like. And, the curve on the top is unwelcome, too.

No oxygen: When there is little oxygen, the stuff should not go woozy. The way it was portrayed in the original game is much more accurate, and much more welcome. The woozy effect should be present with certain medicine, but certainly not zero oxygen.

-------------------------------------------------------------------------------------------------------------------

Problematic bugs:

I was going to add them, but there are too many.

-------------------------------------------------------------------------------------------------------------------

Results: CLUNKY AND TERRIBLE

-------------------------------------------------------------------------------------------------------------------

Suggestions:

Make everything more organized, a bit more empty, and not too clunky.

In addition, to look further into injuries, you click either your own health bar or the other person's health bar.

-------------------------------------------------------------------------------------------------------------------

To be continued.

-------------------------------------------------------------------------------------------------------------------'
Disclaimer: I'm not trying to roast the version, and am trying to improve it and give constructive criticism, as I love this game to death and wouldn't want to play the developer's current version,

-------------------------------------------------------------------------------------------------------------------

Comment away, devs and other bug reporters! 

PS: Update the game in the way that the maker of barotrauma extended updates his version. Please, his way of developing and drastic updates is so much more of my style.

PPS: Loved the bots, hated the way that the chatbox was located. Their random chatter is hilariously awesome!

**Comment by NilanthAnimosus** [comment]
the dev branch is a work in progress and the new graphics felt like a nice yet incomplete (for consistency) change, I even saw a few mockup / temporary graphics in there such as the propeller of the polaris literally has stock image site text on it.

I'm personally not for several aspects myself but I havn't personally played it, only heard of changes, seen screenshots or read some of the commits.

I do not feel the criticism on the graphics is...really the most constructive way to put it. It honestly looks more like a rant - Still there is a mix of old and new graphics that do not quite match up in style (such as the medical boxes vs the other medical objects and lockers). Personally though realistically speaking submarines are cramped spaces with a lot of pipes, equipment and generally compacted space.

I would not call realistic submarines "empty" feeling
![](https://i.ytimg.com/vi/q3suzuLlUoY/maxresdefault.jpg)

Still I consider the dev branch an experimental branch / work in progress, My only immediate thoughts are the inconsistency in wall colours and level of detail makes me feel a bit off, and the locker size feels too much.

I personally think legwear/torso wear and head items should be individual items still, but otherwise cant fully tell how that aspect is.

**Comment by Regalis11** [comment]
A lot of the new things in the dev branch are a work in progress and subject to change, including many of the features and graphics you mentioned. Criticism is more than welcome, especially now when we're still testing out different things and trying to figure out which direction to go to. However, the criticism would be more useful if you were more specific about what's bothering you about some the changes. In any case here's some comments regarding your feedback:

**The player HUD**
I'm assuming this is mostly targeted for the inventory HUD. The main motivation behind the inventory redesign was to reduce the annoying "inventory juggling" - moving items between slots in an emergency situation (especially when you're inventory is nearly full) was slow and clunky and often lead to players suffocating before they managed to get their diving mask on or accidentally dropping items when they're trying to reload a diving suit or a weapon. With the new HUD you only have to click that button above the inventory slot to equip/unequip an item and the things that you rarely need to touch (ID card, headset, uniform) are moved out of the way to the top-right corner. Otherwise I feel it's functionally very close to the old HUD, you can still move items between slots by dragging and double clicking.

One thing that I'm not still sure about is the amount of slots and the "sub inventories" (e.g. the engineers toolbox). The idea was to reduce the amount of always-visible inventory slots and give each class an item where they can hold more stuff, a toolbox for the engineers and mechanics, a weapon holster for the captain, a webbing set for the security and so on. Only the toolbox has been implemented so far, and I'm not that happy with the result as it significantly increases the amount of "inventory juggling" and keeps most of your inventory hidden.

I also agree about the HUD being intrusive, as of now it covers too much of the screen. Being able to hide some of the elements mitigates it to some extent, but I'm not still fully happy with it.

**Medical stuff**
The more detailed health system with limb-specific afflictions has been requested a lot and I think it could make things a lot more interesting. I feel the biggest problems at the moment are the clunky health interface and the lack of content - there aren't many limb-specific afflictions and most of the drugs affect the whole body so the whole feature feels pretty pointless.

**Random HUDs**
This is something I don't fully agree with. I don't see how the extra information (a warning light that indicates whether an item is powered, overvoltage, low voltage, etc) in the interfaces is a bad thing, or how it is better having to click a plus button 20 times instead of just dragging a slider. I agree about the reactor interface though, it's slightly confusing especially now that there's nothing explaining how it works.

**Metal crate**
I'm not sure what you mean by this, how has the metal crate been changed? Do you mean the (WIP) inventory slot sprites that replaced the old black squares?

**New machines**
The new sprites are not final. One of the things the artist who made them was aiming for was to make the submarine feel more cramped and cluttered, with lots of pipes, wires, etc everywhere, much like the pic NilanthAnimosus posted. This was the intention behind making the items larger, to make them fill up more of the sub and reduce the amount of empty space. However I do think it was overdone, some of the items are way too large at the moment and some of the rooms in Polaris (which is a test sub btw) are so cluttered that it's hard to see which parts are decorative and which are something you can interact with.

**Polaris**
A sub mainly for testing the new items and graphics.

**THE INSIDES OF THE STORAGES, THE MENU BUTTONS, JESUS**
What's wrong with these?

**Curved insides**
How/if a sub designer wants to use the sloped walls is up to them. It seems that a lot of people are hacking together "curved walls" by using a bunch of background wall structures, so I think there should be a proper way to make those. They don't have to be used in the way they're used in Polaris, the sloped parts can be made inaccessible from the inside.

**No oxygen**
I think in the old version the effect of drowning/suffocating is not noticeable enough, you can easily get half-way to falling unconscious before you notice that you need oxygen. Now you get a UI indicator when the oxygen level is starting to get low, and a very noticeable (and IMO better looking) effect when you actually start suffocating.

**To look further into injuries, you click your own health bar**
You can already do this

**Comment by BillCombs** [comment]
I looked at the new sprites, and it's a bit concerning for me that some of them seem like they wouldn't fit in small/compact submarines and shuttles. If you are planning to keep them big, then you should also add more compact versions of them(or keep the old ones as a separate option). Otherwise a lot of small subs would most likely look really ugly and unrealistic.

**Comment by Regalis11** [comment]
Yeah, many of them are way too big and will definitely be scaled down. All the old items will be kept in a separate legacy category to maintain backwards compatibility with older subs.

**Comment by AbominationBanisher** [comment]
To respond to Regalis's comment, I'll hopefully explain things in further detail.
-------------------------------------------------------------------------------------------------------------------
- I still have the same feeling about the HUD issue, and I think I have a viable solution. 

- I imagine you making HUD for little problems, kinda like destroying an entire building and building a new one, because the old one's elevator broke.

- Instead, I recommend optimizing the current HUD, or making tiny changes to it. 

- For example, instead of making a new HUD to destroy inventory juggling, you could simply add a tiny little line of code, that makes you equip an item on anything like hands, pants, shirt, mask, and head slots, when you simply --right click-- the item. 

- Want how to see how your health is going? Just simply hover over your own health bar, to get more detailed information. 

- Don't want to die because you don't notice the lack of oxygen? Make the character emit coughing or wheezing sounds, to alert the player of their character's impending suffocation.

-- Tell me any more reasons as to why you'd like the new HUD, and I can recommended a simple optimization to improve the current HUD, so that the new HUD would not be needed. And in addition, I'd find that HUD to be a lot less clunky, and a lot more fun.
-------------------------------------------------------------------------------------------------------------------

- For the medical thing, I have a semi strong opinion on this.

- I still think that the WIP one is  a bit too clunky, although I have a solution.

- The average Joe should not be able to see the in depth menu. Neither should doctors. Only people with the health headset could see the in depth injuries. When you look at someone with injuries through the headset, you get kind of a hologram at the top of the screen, that shows the menu that the WIP one has, but with any optimizations that you might add to the WIP menu. In the hologram, it shows the needed medicine for each body part, and such.

- But, anybody can still simply inject people with medicine. It'll just be less effective without the health headset.

-------------------------------------------------------------------------------------------------------------------
- For the random HUD's, my opinion is gone.

- I just feel like it could possibly be a bit more organized, and less all over the place. You know, a bit more simple.

-------------------------------------------------------------------------------------------------------------------

- For the metal crate, I hold my previous semi opinion.

- I don't like how the toolbox is beside your inventory, taking up more space in the HUD. I feel like you should be able to pick it up, run to anywhere you want, and open the toolbox using the O key. It should not show a sprite of a toolbox beside your inventory IMO. Just keep it like the old version, but to open and close the toolbox it is O. And the menu should be in the center of the screen like the old game, as when you're digging through a toolbox, you shouldn't be able to see everything clearly.

- I hope this kinda makes my previous opinion less confusing.

-------------------------------------------------------------------------------------------------------------------

- For the new sprites, I hold my previous opinion. 

- I honestly feel like the sprites and background textures should be optional, kinda like a texture pack that could be switched in the menu, some texture packs default and others user made.

- For, I honestly don't like clustered textures, although its "realistic" and all. In fact, whenever I'm in a game, I try to put away the stuff others drop on the ground, just because the items make the floor clunky, and make the environment harder to function in.

-------------------------------------------------------------------------------------------------------------------

- For the Polaris, I do agree that it's just a testing place, which I had not realized last time. I think that explains why I didn't really like it.

-------------------------------------------------------------------------------------------------------------------

- For the texture of the menu buttons and all, I was just overreacting (still should be part of a texture pack or something).

-------------------------------------------------------------------------------------------------------------------

- For the curved insides, I generally hold my opinion. But to be honest, there must be a way to better implement the inside curved sprites and the huge tail and front sprite that you displayed on your twitter. I don't know how, but just keep it in mind. 

-------------------------------------------------------------------------------------------------------------------

- For the change in oxygen asthetics, just look at this point that I made when talking about the HUD

- "Don't want to die because you don't notice the lack of oxygen? Make the character emit coughing or wheezing sounds, to alert the player of their character's impending suffocation."

- There, now the new thing is not really useful. But if you like it because of the way it looks, maybe that too, could be part of a texture pack. Who knows?

-------------------------------------------------------------------------------------------------------------------

In conclusion, this is my reply to the developer's points.

Would do you think?

**Comment by Regalis11** [comment]
**HUD**
What I like about the new inventory is that it hides irrelevant information and makes it easier to move items between slots. Equip slots that don't have anything in them are hidden, slots that you rarely need to use are further away from the "main inventory", and you can equip and move items to containers without having to drag them. Ofc previously it was possible to equip or move items by double clicking, but double clicking had two different effects depending on the context which often caused players to accidentally put an item inside a cabinet when they're trying to unequip it.

If we end up using the toolbox, I don't think it's a good idea to have it at the center of the screen. I get your point about not seeing everything when you're digging through the toolbox, but since it's something you have to use fairly often, I feel it should be as non-obstructive as possible. It already takes up too much space on the screen, and having it pop up at the center of the screen would most likely make it even more annoying.

**Lack of oxygen**
There needs to be a very obvious indicator when you're running out of oxygen. I don't think emitting coughing sounds is enough - what if you're playing the without sounds, listening to music while playing or what if there's so much noise (in-game or irl) that you don't notice the sounds?

**Medical**
If you need a health scanner to see the new health interface, I think we'd have two options: either we make it possible to heal most injuries without having to apply a treatment to a specific limb, or having a health scanner becomes essentially mandatory to heal characters effectively. I don't see either of these as good options, the first one makes the limb-specific health system pretty much pointless and the second means that you should always be carrying a health scanner.

**New sprites**
I agree with you about the textures being too cluttered, they make it too difficult to see what's interactable and what's just for decoration. For example in Polaris the engine room is just a mess. All of the new item/structure sprites will be improved and I hope that most sub creators will start using them, but as said the old ones will also be kept as a legacy option.

**Curved insides**
I've been thinking of different ways to implement the curved walls, and in addition to the current one I see a few options. The submarine could have an interior hull which consists of mostly rectangular rooms, and the curved pieces could be used as the exterior hull which can be reached via a small "service space" between the hulls. Another way would be to add sloped wall sprites with no collision detection (much like the DIY sloped walls in many custom subs).

The enormous tail and bow sprites are not going to be kept in their current state, although I'm not completely sure yet what to do about them. They could be changed to decorative background sprites with no collision detection, but I'm not too happy about that idea because at least personally I would find it strange that the bow of the submarine can go through walls.

**Comment by JJRcop** [comment]
My views, if they are worth anything:

I like the new suffocation effect
I like the new HUD, even if it's still being worked out. It could be a little smaller tho, or hidable.
I like the new sprites for the equipment, even if they are still WIP.
I really like the new health system, and reactor UI. An in-game manual item could help with the confusion.
I'm a huge fan of less inventory slots by default, and getting more space by carrying around a toolbox or other container.

**Comment by Billsofat** [comment]
Well, I guess I'll offer my opinions on this.

**Inventory**
So far, I find the old one easier and quicker to use. By being in the lower left corner, it was very compact and easy to move things around. With some slots being in the upper right, it makes me look around the screen more to manage my items, and it feels less organized.

I find that the toolbox makes it harder to manage your inventory. Rather than having 10 slots and a jumpsuit right next to them, you have search through a huge amount of slots (which would take longer), and it blocks your view. Having 5 slots instead of 10 also hurts doctors and security a lot. Especially considering doctors have to take their uniform off in order to access the few slots in it, which would slow them down a lot.

The new one also takes up way too much space. The sliding partially out of view just feels clunky, and adds a whole extra delay to access the inventory. I personally don't find that the old "juggling" takes very much time, and I feel like the buttons just make it harder to organize and see what's equipped where. With the 2x5 item slots, I could organize my items. Now, I can't really do that.

**Medical**
I like the idea behind the new system, but there are a few things that bother me:

It takes a long time to look through the body part display and dragging syringes over to it just feels like adds a lot of more "searching time" in order to drag the right thing. I feel like this would be too slow, especially in emergency situations. Medicine seems harder to move around in the inventory too.

For how quickly one can get injured in the game, I just feel like healing would be too long and cumbersome of a process, especially considering CPR is pretty much murder in the dev build (unless you're a doctor in which case it seems pretty much instant). Medicine is also harder to move around in the inventory.

Overall, I have mixed feelings about it. It's cool, but I think simplicity isn't always a bad thing.

**Singleplayer AI management**
The commands are a cool system, but the huge list of them just seems too long and cumbersome to look through. Maybe have a few commands that you can 'pin',, and then a menu you can open up to see the other ones.

**New sprites and tightness/emptiness**
I agree that they are far too big, especially the engine and pumps. I also don't like the art style of the new ones. They just seem to complicated and take long to look at. With the old art, they are immediately recognizable, and easy to see and organize in the room, since most of them rectangular.

As far as the backgrounds and hulls go, I'm not sure I like the idea of giant hull sprites, like the tail and front on the polaris. It seems like a drastic change in the idea of submarine's in the game, since in the current versions, your room design determines everything about the sub. Now, if all of them just fit inside a realistic shape of tail and front, it seems like an element of creativity is removed, as if part of the sub is built for you. I somewhat feel that way about curved walls, as there were interesting ways of fitting equipment in the jagged walls, wheres nothing really fits in the curved walls.

Also, on the polaris, it looks weird having room sized propellers way far out back when the engine is way back inside the sub. Again, I know it is a work in progress.

About the reality and tightness of the subs, I feel like having it look as cramped as real life may be a bit too realistic for a 2D game. For one, this game takes place hundreds of years in the future. Equipment size could probably be scaled down by then. 

Second, part of the point is that it _is_ a game. Things almost always aren't as complicated as in real life, where you have to go through months/years of training to understand and be qualified to work in a submarine.

**Low oxygen effect**
I never really thought the old one was hard to notice, but I understand your point. I just feel like the new one is a bit too distracting.

**Comment by NilanthAnimosus** [comment]
I'm with billsofat on the inventory - If it wasn't for slot issues where the expansion of a containers slots is cancelled as you are trying to use it wasn't a thing I actually liked the compact nature of the inventory being at a single side of the screen.

I am totally for the new sprites the Polaris has on the rear/front. But only if sub creators are allowed to make them optional and also add their own as part of the submarines design (Sort of like uploading a custom sprite into the submarine file for that sub) with sprites being available as part of the content pack (Or even not required to be in the content pack itself - and instead included in submarine downloads to the content folder of sorts if a submarine has made use of it).

**Comment by AbominationBanisher** [comment]
Billsofat just explained most of the things I wanted to say, but in a generally better way. Glad to see that someone holds the same opinion as me.

**Comment by Regalis11** [comment]
Thanks for the extra feedback everyone! Again, some comments:

**Inventory**
Regarding the toolbox, I fully agree with Billsofat. The motivation behind it was to hide items that you're not using frequently and only have a few slots for the "active inventory" (5 slots is probably too little for that though), but it seems it only made the problem we're trying to solve worse. As a side note, the idea was to give the rest of the classes a similar item too, not to restrict their inventory capacity to 5. The inventory slots inside uniforms and jumpsuits were going to be removed, so all the slots in the upper right corner would've been something you very rarely need to touch and having them out of the way was intended to make organizing the "active inventory" easier. In any case I feel like the inventory should go back on the drawing board.

**Medical**
I agree that the more complicated medical system and UI is too slow considering how fast characters can die. I think it's a good thing that you have to pay some more thought on how to heal a character, but when a character can bleed out or suffocate in a matter of seconds there's no time to do that. But I feel this is not a super difficult problem to solve, with a more streamlined UI and slowing down the effects of the afflictions I think we can make it work way better. And yeah, CPR is unbalanced as hell at the moment.

**Singleplayer AI management**
I think one way to make the UI easier to use would be to group the icons according to how appropriate the command is for a given class (e.g. captain has the steering command right next to the icon). One group for commands that are meant for that specific class, another group for non-class-specific commands and a third one for the ones the class should not be doing. The third and maybe the second group could be hidden by default.

The icons also need to be replaced with way more recognizable ones, now many of them are placeholders that have nothing to do with the command.

**New sprites**
I'm not a fan of the huge tail and bow sprites either, they restrict the shape of the submarine too much and now without any collision detection they don't work at all gameplay-wise. If they are kept in some form they need to be more modular and there needs to be multiple variants to make them fit on different kinds of subs. I get Billsofat's point about the curved walls, it's hard to make any use of the curved spaces. But I think the new walls are still a useful thing to have - many sub creators are already hacking together similar ones from multiple wall pieces and it's good to have a proper alternative for those. And as said they don't have to be used in the way they're used in Polaris; you could for example have a few decorative slopes here and there and keep the space between the inner hull and the slope as an empty "service space". Or just leave them out completely and make the subs the same way as before.
